src/java/org/apache/nutch/parse/ParseOutputFormat.java

```diff
@@ -362,7 +362,6 @@ public static String filterNormalize(String fromUrl, String toUrl,
       if (ignoreExternalLinks) {
         if ("bydomain".equalsIgnoreCase(ignoreExternalLinksMode)) {
           String toDomain = URLUtil.getDomainName(targetURL).toLowerCase();
-          //FIXME: toDomain will never be null, correct?
```

Comment:
Please check whether URLUtil.getDomainName(URL url) may return null and add this information to the javadoc of the function. The FIXME should then be resolved accordingly.
If you're uncertain about the behavior, you could a test to TestURLUtil.testGetDomainDame() to test the actual behavior, e.g.,
```
    // test URL without host/authority
    url = new URL("file:/path/index.html");
    Assert.assertNull(URLUtil.getDomainName(url));
    Assert.assertNotNull(URLUtil.getDomainName(url));
    Assert.assertEquals("", URLUtil.getDomainName(url));
```
Of course, not all three assertions may succeed. Please also add such a test to your PR, or even more for edge cases of URLs without host or domain.